### ISSUE TITLE
A11y Followup Based on LH

### ISSUE THREAD

**Issue by ColinBuyck** [bloom-housing/bloom#4828]
Emily and I have been discussing the use of the grid component in the new filter drawer and the impact it has on SR. In short, the layout itself communicates everything that a table would which obscures the form announcements. The online documentation around the issue is unclear so I will reach out to LH and implement their feedback

**UPDATE**
The initial description of this ticket is below. We have fixed all the grid issues, but I want to confirm still if this excerpt is an issue: The fieldset labels are not announced when navigating between fieldsets using just tabs, in VoiceOver; on JAWS it is announced. User should know which field group (fieldset) they’re in. See the following ling for examples (https://www.w3.org/WAI/tutorials/forms/grouping/)

<hr />

LH Audit Results:
Please refer to the WAI ARIA Guide on Grid (Layout Container) Pattern (https://www.w3.org/WAI/ARIA/apg/patterns/grid/#layoutgridsforgroupingwidgets). I believe you’re interested in avoiding the table-like experience and instead have a form group-like experience. One basic issue is that inside the fieldset (grid), pressing tab does not move to the next fieldset (grid) and instead moves to the next element inside the fieldset. This is not expected. Also, one cannot use the arrow keys to move between fields inside the fieldset (grid), and instead the page scrolls. 

The expectation is that the arrows let user move the focus inside the grid (fieldset).
While the choice of using an ARIA grid to afford arrow key navigation within the grid, it doesn’t seem to work. There’s also the negative effect of JAWS and VO calling out the entry into and exit from the tables, which is confusing, as you had alluded to. Was there any other reason for using ARIA grids other than visual, for which CSS grids might be more suitable?

There is a symmetry between using the role=”grid” (https://www.w3.org/TR/wai-aria-1.1/#grid) and role=”gridcell” (https://www.w3.org/TR/wai-aria-1.1/#gridcell). It seems that you’re using role=”grid” and role=”cell”. This might be the reason for the breakage of the arrow keys usage.
Tabbing through the groups and/or controls sometimes obscures the focus fully or partially. The filters are not scrolled so that the focus remains visible.

The fieldset labels are not announced when navigating between fieldsets using just tabs, in VoiceOver; on JAWS it is announced. User should know which field group (fieldset) they’re in. See the following ling for examples (https://www.w3.org/WAI/tutorials/forms/grouping/)

**Comment by emilyjablonski** [comment]
This also affects partners

**Comment by ColinBuyck** [comment]
@sarahlazarich Great news! Did some testing on detroit dev and all of LH's concerns are now resolved thanks to the work done earlier by @emilyjablonski  [here](https://github.com/bloom-housing/ui-seeds/pull/125). So this is fixed and ready for launch 🌵
